src/components/menu/components/menuLinks/MenuLinks.js

```diff
@@ -0,0 +1,70 @@
+//Libraries
+import React from 'react';
+import PropTypes from 'prop-types';
+
+//Helpers
+import _map from 'lodash/map';
+import { isEmpty } from 'lodash';
+
+//constants
+import { INITIAL_STATE } from './constants/menuLinks.general';
+
+//Components
+import LinkItem from '../../../../commonComponents/linkItem';
+
+//CSS
+import './menuLinks.css';
+
+class MenuLinks extends React.Component {
+  state = INITIAL_STATE;
+
+  componentDidMount() {
+    const {
+      location: { hash },
+    } = window;
+    const { categories } = this.props;
+    const currActive = isEmpty(hash) ? categories[0].name : hash.slice(1);
+    this.setState({
+      currActive: currActive,
+    });
+  }
+
+  setCurrentActiveLink = (newRef) => {
+    const newActive = newRef.current.hash.substring(1);
+    this.setState({ currActive: newActive });
+  };
+
+  renderMenuLinkItem = (category) => {
+    const { currActive } = this.state;
+    const isEqual = category === currActive;
+    const linkRef = React.createRef();
```

Comment:
you are creating a ref for every link item and on click of that link you are getting hash property of that link
since the hash of anchor tag is created by href and href is created by category variable
you can just call the onclick using the category in link item instead of using ref here
this will also save you from using .substring method in line 33

Reply:
Again not reuasble